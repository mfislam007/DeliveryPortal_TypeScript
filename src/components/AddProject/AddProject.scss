.icon-image,
.text-field {
  margin-top: 20px;
  margin-left: 20px;
}
