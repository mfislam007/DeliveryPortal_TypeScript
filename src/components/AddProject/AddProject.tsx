import React, { useState, useEffect } from "react";

import { makeStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import IconButton from "@material-ui/core/IconButton";
import Typography from "@material-ui/core/Typography";
import CloseIcon from "@material-ui/icons/Close";
import Slide, { SlideProps } from "@material-ui/core/Slide";
import AddIcon from "@material-ui/icons/Add";
import TextField from "@material-ui/core/TextField";
import Grid from "@material-ui/core/Grid";
import AccountCircle from "@material-ui/icons/AccountCircle";
import ViewListIcon from "@material-ui/icons/ViewList";
import { useHistory } from "react-router-dom";

import { addProject } from "../../controllers/ProjectController/ProjectController";

const templates = [
  {
    value: "Prima-Power Project",
    label: "Prima-Power Project",
  },
  {
    value: "Nokia Project",
    label: "Nokia Project",
  },
  {
    value: "Huawei project",
    label: "Huawei project",
  },
  {
    value: "Sony Project",
    label: "Sony Project",
  },
  {
    value: "ABB Project",
    label: "ABB Project",
  },
];

type Props = {
  projectName: string;
  customerName: string;
  managerName: string;
  projectTemplate: string;
};

const useStyles = makeStyles(theme => ({
  appBar: {
    position: "relative",
  },
  title: {
    marginLeft: theme.spacing(2),
    flex: 1,
  },
  button: {
    margin: theme.spacing(1),
  },
}));

const Transition = React.forwardRef<HTMLAnchorElement, SlideProps>(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
});

const AddProject: React.FC = (): JSX.Element => {
  let history = useHistory();
  const classes = useStyles();
  const [open, setOpen] = useState(false);
  const [project, setProject] = useState<Props>({
    projectName: "",
    customerName: "",
    managerName: "",
    projectTemplate: "Prima-Power Project",
  });

  useEffect(() => {
    localStorage.setItem("project", JSON.stringify(project));
  }, [project]);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const onChangeProject = (event: React.ChangeEvent<HTMLInputElement>) => {
    switch (event.currentTarget.id) {
      case "projectName":
        let tempProject = { ...project };
        tempProject.projectName = event.currentTarget.value;
        setProject(tempProject);
        setProject({
          projectName: event.currentTarget.value,
          customerName: project.customerName,
          managerName: project.managerName,
          projectTemplate: project.projectTemplate,
        });
        break;
      case "customerName":
        setProject({
          projectName: project.projectName,
          customerName: event.currentTarget.value,
          managerName: project.managerName,
          projectTemplate: project.projectTemplate,
        });
        break;

      case "managerName":
        setProject({
          projectName: project.projectName,
          customerName: project.customerName,
          managerName: event.currentTarget.value,
          projectTemplate: project.projectTemplate,
        });
        break;

      case "projectTemplate":
        setProject({
          projectName: project.projectName,
          customerName: project.customerName,
          managerName: project.projectName,
          projectTemplate: event.currentTarget.value,
        });
        break;
    }
  };

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    {
      alert(
        " Project Name: " +
          project.projectName +
          " , " +
          " Customer Name: " +
          project.customerName +
          " , " +
          " ProjectManager Name: " +
          project.managerName +
          " , " +
          " Project Template: " +
          project.projectTemplate +
          " are created "
      );
    }
    addProject(
      project.projectName,
      project.customerName,
      project.managerName,
      project.projectTemplate
    );
    event.preventDefault();
    setOpen(false);
    history.push("/deliveryportal");
  };

  const enabled =
    project.projectName.length > 3 &&
    project.customerName.length > 3 &&
    project.managerName.length > 3;

  return (
    <div>
      <IconButton
        color="inherit"
        aria-label="add to cart"
        onClick={handleClickOpen}
        style={{ marginTop: 0 }}
      >
        <AddIcon />
      </IconButton>

      <div>
        <Dialog fullScreen open={open} onClose={handleClose} TransitionComponent={Transition}>
          <AppBar className={classes.appBar}>
            <Toolbar>
              <IconButton edge="start" color="inherit" onClick={handleClose} aria-label="close">
                <CloseIcon />
              </IconButton>
              <Typography variant="h6" className={classes.title}>
                Add Delivery Project
              </Typography>
              <div>
                <Button autoFocus color="inherit" type="submit" form="my-form" disabled={!enabled}>
                  save
                </Button>
              </div>
            </Toolbar>
          </AppBar>

          <form onSubmit={handleSubmit} id="my-form">
            <div>
              <Grid container spacing={1} alignItems="flex-end">
                <Grid item>
                  <AccountCircle style={{ marginTop: 20, marginLeft: 20 }} />
                </Grid>
                <Grid item xs={6}>
                  <TextField
                    id="projectName"
                    label="Project Name"
                    style={{ marginTop: 20, marginLeft: 20 }}
                    onChange={onChangeProject}
                    fullWidth
                  />
                </Grid>
              </Grid>
            </div>
            <div>
              <Grid container spacing={1} alignItems="flex-end">
                <Grid item>
                  <AccountCircle style={{ marginTop: 20, marginLeft: 20 }} />
                </Grid>
                <Grid item xs={6}>
                  <TextField
                    id="customerName"
                    label="Customer Name"
                    style={{ marginTop: 20, marginLeft: 20 }}
                    onChange={onChangeProject}
                    fullWidth
                  />
                </Grid>
              </Grid>
            </div>
            <div>
              <Grid container spacing={1} alignItems="flex-end">
                <Grid item>
                  <AccountCircle style={{ marginTop: 20, marginLeft: 20 }} />
                </Grid>
                <Grid item xs={6}>
                  <TextField
                    id="managerName"
                    label="ProjectManager Name"
                    style={{ marginTop: 20, marginLeft: 20 }}
                    onChange={onChangeProject}
                    fullWidth
                  />
                </Grid>
              </Grid>
            </div>
            <div>
              <Grid container spacing={1} alignItems="flex-end">
                <Grid item>
                  <ViewListIcon style={{ marginTop: 20, marginLeft: 20 }} />
                </Grid>
                <Grid item xs={6}>
                  <TextField
                    id="projectTemplate"
                    select
                    label="Project Template"
                    value={project.projectTemplate}
                    onChange={onChangeProject}
                    fullWidth
                    style={{ marginTop: 20, marginLeft: 20 }}
                    SelectProps={{
                      native: true,
                    }}
                  >
                    {templates.map(option => (
                      <option key={option.value} value={option.value}>
                        {option.label}
                      </option>
                    ))}
                  </TextField>
                </Grid>
              </Grid>
            </div>
          </form>
        </Dialog>
      </div>
    </div>
  );
};
export default AddProject;
