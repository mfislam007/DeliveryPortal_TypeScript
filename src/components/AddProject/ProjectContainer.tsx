import React from "react";
interface Props {
  project: { name: string; customer: string; ProjectManager: string };
}

const ProjectContainer: React.FC<Props> = (props): JSX.Element => {
  return <div>{props.project.name}</div>;
};

export default ProjectContainer;
