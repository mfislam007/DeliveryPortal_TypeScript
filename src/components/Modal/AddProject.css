.content label {
  padding: 5px 5px;
  margin-right: 50px;
  font-size: 18px;
}
.content input {
  padding-left: 5px;
  width: 15rem;
  font-size: 18px;
}

.button1 {
  background-color: white;
  border: none;
  color: black;
  padding: 10px 0px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 15px;
  margin: 10px 10px;
  cursor: pointer;
}
.button2 {
  background-color: white;
  border: none;
  color: black;
  padding: 10px 30px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 15px;
  margin: 10px 10px;
  cursor: pointer;
  border: blanchedalmond;
}
.maincontent {
  text-align: center;
}
