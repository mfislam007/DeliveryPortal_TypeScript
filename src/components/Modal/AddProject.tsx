import React, { useState, useEffect } from "react";
import Modal from "react-modal";
import "./AddProject.css";

const AddProject: React.FC = (): JSX.Element => {
  const [modalIsOpen, setModalIsOpen] = useState(false);
  const [projectName, setProjectName] = useState("");
  const [customerName, setCustomerName] = useState("");

  useEffect(() => {
    setProjectName(JSON.parse(localStorage.getItem("projectName")));
    if (!localStorage.getItem("projectName")) {
      setProjectName("projectName");
    } else {
      setProjectName("");
    }
  }, []);

  useEffect(() => {
    setCustomerName(JSON.parse(localStorage.getItem("customerName")));
    if (!localStorage.getItem("customerName")) {
      setCustomerName("customerName");
    } else {
      setCustomerName("");
    }
  }, []);

  useEffect(() => {
    localStorage.setItem("projectName", JSON.stringify(projectName));
  }, [projectName]);

  useEffect(() => {
    localStorage.setItem("customerName", JSON.stringify(customerName));
  }, [customerName]);

  const onRequestOpen = () => {
    setModalIsOpen(true);
  };
  const onRequesClose = () => {
    setModalIsOpen(false);
  };

  const onChangeProject = (event: React.FormEvent<HTMLInputElement>) => {
    setProjectName(event.currentTarget.value);
  };

  const onChangeCustomer = (event: React.FormEvent<HTMLInputElement>) => {
    setCustomerName(event.currentTarget.value);
  };

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    alert("project name: " + projectName + " and " + "customer name: " + customerName);
    setProjectName("");
    setCustomerName("");
    event.preventDefault();
  };

  const isEnabled = projectName.length > 3 && customerName.length > 3;

  return (
    <div className="maincontent">
      <button onClick={onRequestOpen}>Open modal</button>
      <Modal isOpen={modalIsOpen}>
        <form onSubmit={handleSubmit}>
          <div className="projectcontent">
            <h2 text-align="center">Add Delivery Portal</h2>

            <label>Project Name:</label>
            <input value={projectName} type="text" onChange={onChangeProject} />
          </div>
          <div className="customercontent">
            <label>Customer Name:</label>
            <input value={customerName} type="text" onChange={onChangeCustomer} />
          </div>
          <div>
            <button className="button1" onClick={onRequesClose}>
              тип Cancel
            </button>
            <button className="button2" type="submit" disabled={!isEnabled}>
              + Create
            </button>
          </div>
        </form>
      </Modal>
    </div>
  );
};
export default AddProject;
