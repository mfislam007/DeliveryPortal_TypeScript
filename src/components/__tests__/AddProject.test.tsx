import React from "react";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { describe, expect, it } from "@jest/globals";

import AddProject from "../AddProject/AddProject";

configure({ adapter: new Adapter() });
describe("Testing for AddProject component", () => {
  it("Dialog exists", () => {
    const wrapper = shallow(<AddProject />);
    expect(wrapper.find("Dialog")).toMatchSnapshot();
  });
  it("Dialog has form", () => {
    const wrapper = shallow(<AddProject />);
    expect(wrapper.find("form")).toMatchSnapshot();
  });
  it("Dialog has taxtfields", () => {
    const wrapper = shallow(<AddProject />);
    expect(wrapper.find("TextField")).toMatchSnapshot();
  });
  it("Dialog has buttons", () => {
    const wrapper = shallow(<AddProject />);
    expect(wrapper.find("Button")).toMatchSnapshot();
  });
});
