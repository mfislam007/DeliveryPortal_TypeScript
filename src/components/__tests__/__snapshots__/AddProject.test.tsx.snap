// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for AddProject component Dialog buttons 1`] = `ShallowWrapper {}`;

exports[`Testing for AddProject component Dialog exists 1`] = `ShallowWrapper {}`;

exports[`Testing for AddProject component Dialog has buttons 1`] = `ShallowWrapper {}`;

exports[`Testing for AddProject component Dialog has form 1`] = `ShallowWrapper {}`;

exports[`Testing for AddProject component Dialog has four taxtfields 1`] = `ShallowWrapper {}`;

exports[`Testing for AddProject component Dialog has one form 1`] = `ShallowWrapper {}`;

exports[`Testing for AddProject component Dialog has taxtfields 1`] = `ShallowWrapper {}`;

exports[`Testing for AddProject component Dialog has two buttons 1`] = `ShallowWrapper {}`;
